### PR TITLE
[#157969241] Add session endpoint to swagger specs.

### PR DESCRIPTION
**Pull request by lussoluca**
Also move `/session` to `/api/v1/session` and use the generated `PublicSession`

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#157969241](https://www.pivotaltracker.com/story/show/157969241): Aggiungere l'endpoint `/session` alle spec del proxy

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: api_proxy.yaml [hunk 1]
```diff
@@ -393,6 +416,22 @@ definitions:
       - name
       - preferred_email
       - version
+  PublicSession:
+    type: object
+    title: User session data
+    description: Describe the current session of an authenticated user.
+    properties:
+      expired:
```
Comment (cloudify) [inline api_proxy.yaml]: why we have both `expired` and `expireAt`, don't they tell the same thing?

File: api_proxy.yaml [hunk 2]
```diff
@@ -393,6 +416,22 @@ definitions:
...
+        type: boolean
+      expireAt:
+        type: number
+      newToken:
+        type: string
+      spidLevel:
```
Comment (cloudify) [inline api_proxy.yaml]: this should probably be an enum

File: api_proxy.yaml [hunk 3]
```diff
@@ -393,6 +416,22 @@ definitions:
...
       - version
+  PublicSession:
+    type: object
+    title: User session data
+    description: Describe the current session of an authenticated user.
+    properties:
+      expired:
+        type: boolean
+      expireAt:
+        type: number
```
Comment (cloudify) [inline api_proxy.yaml]: this should be an ISO timestamp

Comment (gunzip) [reply api_proxy.yaml]: ref. https://github.com/teamdigitale/digital-citizenship-functions/blob/master/api/definitions.yaml#L428

File: api_proxy.yaml (hunk 2 above)
Comment (gunzip) [reply api_proxy.yaml]: ref. https://github.com/teamdigitale/digital-citizenship-functions/blob/master/api/definitions.yaml#L37

File: api_proxy.yaml (hunk 1 above)
Comment (lussoluca) [reply api_proxy.yaml]: @fpersico do you remember why the app needs both?

File: api_proxy.yaml (hunk 2 above)
Comment (lussoluca) [reply api_proxy.yaml]: The problem here is that the possible keys are in the format: `https://www.spid.gov.it/SpidL1` so the generated enum will be:

```
export enum SpidLevelEnum {
  "https://www.spid.gov.it/SpidL1" = "https://www.spid.gov.it/SpidL1",

  "https://www.spid.gov.it/SpidL2" = "https://www.spid.gov.it/SpidL2",

  "https://www.spid.gov.it/SpidL3" = "https://www.spid.gov.it/SpidL3"
}
```

and `SpidLevelEnum.https://www.spid.gov.it/SpidL1` is not valid. Or there is a different way to reference those enums?

File: api_proxy.yaml (hunk 3 above)
Comment (lussoluca) [reply api_proxy.yaml]: @fpersico do you see any problem in switching from a unix epoch timestamp to a ISO timestamp in the app?

Comment (gunzip) [reply api_proxy.yaml]: btw swagger specs supports only date-time = ISO time

File: api_proxy.yaml (hunk 2 above)
Comment (gunzip) [reply api_proxy.yaml]: SpidLevelEnum["https://www.spid.gov.it/SpidL1"]

File: api_proxy.yaml (hunk 1 above)
Comment (fpersico) [reply api_proxy.yaml]: I don't think it is a requirement, expireAt is enough.

File: api_proxy.yaml (hunk 3 above)
Comment (fpersico) [reply api_proxy.yaml]: isostring is good

File: src/services/redisSessionStorage.ts [hunk 4]
```diff
@@ -108,8 +107,10 @@ export default class RedisSessionStorage implements ISessionStorage {
       );
     }
 
+    const expire = timestamp + this.tokenDurationSecs * 1000;
```
Comment (cloudify) [inline src/services/redisSessionStorage.ts]: expire -> expireAtEpochMs

**Pull request merged**